### PR TITLE
[Bugfix] Fix issue with saving example text assessments

### LINKED ISSUE #2500: [Bug] Text example assessment cannot be created
### Describe the bug
When instructors try to add an example assessment of example submissions for text exercises, a `500` server error is thrown. This is because, the _create new assessment_ button calls the create assessment route which was refactored in #2438 for re-use. It now expects the `submission.studentParticipation` to be set (which is not the case for example submissions). When it tries to call `studentParticipation.getId()` the following error is thrown:

`NullPointerException
Cannot invoke "de.tum.in.www1.artemis.domain.participation.StudentParticipation.getExercise()" because "studentParticipation" is null`, see: https://sentry.io/organizations/ls1intum/issues/2047241553/?environment=test&project=1440029&query=is%3Aunresolved

#### To Reproduce
1. Go to 'Course Management'
2. Click on 'Edit a text exercise'
3. Scroll down to 'Create an example submission'
4. Add an example assessment
5. Create example assessment
6. observe error in console

#### Expected behavior
The example assessment should be saved correctly. 

### Environment
<details><pre>
 - OS: [Windows]
 - Browser [Brave, chromium]
 - Version [Version 1.17.73 Chromium: 87.0.4280.67 (Official Build) (64-bit)]
</pre></details>

#### Output in Web Console
image
image

### PR DESCRIPTION
**Pull request by fde312**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

Fixes #2500 

### Description

- Extracted Resource call to parent
- Adapted children accordingly
- Use new endpoint for saving example assessment for text exercise

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new text exercise or use an old one and edit it
4. Create an example submission
5. Create a example assessment and click on update assessment

### Screenshots

image

image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java [hunk 1]
```diff
@@ -107,6 +112,23 @@ public TextAssessmentResource(AuthorizationCheckService authCheckService, TextAs
...
+     * @return result after saving example text assessment
+     */
+    @ResponseStatus(HttpStatus.OK)
+    @ApiResponses({ @ApiResponse(code = 403, message = ErrorConstants.REQ_403_REASON), @ApiResponse(code = 404, message = ErrorConstants.REQ_404_REASON) })
+    @PutMapping("/text-submissions/{submissionId}/example-assessment")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Result> saveTextExampleAssessment(@PathVariable long submissionId, @RequestBody List<Feedback> feedbacks) {
+        log.debug("REST request to save text example assessment : {}", submissionId);
+        // TODO: Figure out if we need to use saveTextBlocks
+        return super.saveExampleAssessment(submissionId, feedbacks);
```
Comment (fde312) [inline src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java:129]: @jpbernius I am not quite sure if we need to use `saveTextBlocks` here. For the example assessment, we are using the old version of assessing text exercises

File: src/main/webapp/app/exercises/text/manage/example-text-submission/example-text-submission.component.ts [hunk 2]
```diff
@@ -350,12 +350,12 @@ export class ExampleTextSubmissionComponent implements OnInit, AfterViewInit {
...
             return;
         }
-
-        this.assessmentsService.save(this.exercise.id!, this.result.id!, this.assessments, []).subscribe((response) => {
+        this.assessmentsService.saveExampleAssessment(this.assessments, this.exampleSubmission.id!).subscribe((response) => {
             this.result = response.body!;
             this.areNewAssessments = false;
             this.jhiAlertService.success('artemisApp.textAssessment.saveSuccessful');
         });
+        // this.assessmentsService.save(this.exercise.id!, this.result.id!, this.assessments, [])
```
Comment (fde312) [inline src/main/webapp/app/exercises/text/manage/example-text-submission/example-text-submission.component.ts:358]:
```suggestion

```

File: src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java (hunk 1 above)
Comment (fde312) [reply src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java:129]: We decided that it is not needed as the client would send an empty array to the server anyways.

**Review by airelawaleria — APPROVED** [review]
Tested in the testing session. Works fine

**Review by JohannesWeiss — APPROVED** [review]
Tested in testing session, works as intended.

**Review by simonzilker — APPROVED** [review]
Tested with @airelawaleria @JohannesWeiss @marvinchimeka on ts1. Everything works as intended!

**Review by marvinchimeka — APPROVED** [review]
Tested in a testing session. Approve.

**Review by JannisBittlmayer — APPROVED** [review]
Tested on TS3, works as intended

**Review by FlorianStang — APPROVED** [review]
Tested on TS3 everything worked as intended

**Review by EneaGore — APPROVED** [review]
Works as intended. Approve.

**Review by krusche — APPROVED** [review]

**Pull request merged**